tests/components/enphase_envoy/test_sensor.py

```diff
@@ -843,3 +844,62 @@ def integration_disabled_entities(
         )
         if entity_entry.disabled_by == er.RegistryEntryDisabler.INTEGRATION
     ]
+
+
+@pytest.mark.parametrize(
+    ("mock_envoy"),
+    [
+        "envoy_metered_batt_relay",
+    ],
+    indirect=["mock_envoy"],
+)
+@pytest.mark.usefixtures("entity_registry_enabled_by_default")
+async def test_sensor_missing_data(
+    hass: HomeAssistant,
+    config_entry: MockConfigEntry,
+    mock_envoy: AsyncMock,
+    entity_registry: er.EntityRegistry,
+) -> None:
+    """Test enphase_envoy sensor platform midding data handling."""
+    with patch("homeassistant.components.enphase_envoy.PLATFORMS", [Platform.SENSOR]):
+        await setup_integration(hass, config_entry)
+
+    ENTITY_BASE = f"{Platform.SENSOR}.envoy_{mock_envoy.serial_number}"
+    coordinator: EnphaseUpdateCoordinator = hass.data[DOMAIN][config_entry.entry_id]
+
+    # force missing data to test 'if == none' code sections
+    mock_envoy.data.system_production_phases["L2"] = None
+    mock_envoy.data.system_consumption_phases["L2"] = None
+    mock_envoy.data.ctmeter_production = None
+    mock_envoy.data.ctmeter_consumption = None
+    mock_envoy.data.ctmeter_storage = None
+    mock_envoy.data.ctmeter_production_phases = None
+    mock_envoy.data.ctmeter_consumption_phases = None
+    mock_envoy.data.ctmeter_storage_phases = None
+
+    # use different inverter serial to test 'expected inverter missing' code
+    mock_envoy.data.inverters["2"] = mock_envoy.data.inverters.pop("1")
+
+    # force HA to detect changed data by changing raw
+    mock_envoy.data.raw = {"I": "am changed"}
+
+    await coordinator.async_request_refresh()
```

Comment:
Instead of directly interacting with internal parts (in this case the coordinator), we should move time forward (the next scan interval).